ODIN_II/SRC/OdinGridAnalyzer.cpp

```diff
@@ -0,0 +1,334 @@
+	std::vector<std::vector<std::pair<char*,int>>> grid (grid_height,std::vector<std::pair<char*,int>>(grid_width,std::make_pair(nullptr,-1)));
+	std::vector<t_grid_loc_def>::iterator pointer;
+	std::string typeTag = getArchDescriptionTag(hardBlockType);
+	std::cout<<typeTag<<std::endl;
+	int priorityOfTheDesiredBlock = findDesiredBlockPriority(layout,typeTag);
+	std::cout<<"Desired block's priority is: "<<priorityOfTheDesiredBlock<<std::endl;
+	std::cout<<"BlockDefsAre:"<<std::endl;
+	for (pointer =  layout.loc_defs.begin();pointer<layout.loc_defs.end(); pointer++){
+		// Case of == is the tile type we are looking for, and a clash
+		// if there are collisions, vpr would fail anyway
+		if (priorityOfTheDesiredBlock <= pointer->priority){
+			std::cout<<pointer->block_type<<":"<<std::endl;
+			ssize_t indexOfTile  = findIndexOfAssociatedTileType(tileTypes,pointer->block_type);
+			if (indexOfTile == -1)
+				{
+					std::cout<<" Was not able to find a tile"<<std::endl;
+					flush(std::cout);
+					exit(6);
+				}
+			t_physical_tile_type& tileTypeOfABlockWithHigherPriority = tileTypes[indexOfTile];
+			int w = 0;
+			int h = 0;
+			std::cout<<"Width of a block is:\t"<<tileTypeOfABlockWithHigherPriority.width<<"\tHeight of a block is:"
+				<<tileTypeOfABlockWithHigherPriority.height<<std::endl;
+			w = tileTypeOfABlockWithHigherPriority.width;
+			h = tileTypeOfABlockWithHigherPriority.height;
+		
+			fillGridWithBlockType(grid,&(*pointer),tileTypes,indexOfTile,grid_width,grid_height);
+		}
+	}
+	std::cout<<"Grid is:"<<std::endl;
+	for (size_t x = 0; x<grid_width; x += 1) 
+	{
+	for (size_t y = 0; y <grid_height; y += 1) 
+		{
+		if (grid[x][y].first != nullptr){
+			std::cout<<"("<<grid[x][y].first<<","<<grid[x][y].second<<")\t";
+		} else {
+			std::cout<<"( u, -1), ";
+		}
+		}
+	std::cout<<std::endl;
+	}
+	int count = 0;
+	std::cout<<"Grid is:"<<std::endl;
+	for (size_t x = 0; x<grid_width; x += 1) 
+	{
+	for (size_t y = 0; y <grid_height; y += 1) 
+		{
+		if (grid[x][y].first != nullptr){
+			std::string stringToCompare = std::string(grid[x][y].first);
+			if (stringToCompare.compare(typeTag)==0)
+				count = count + 1;
+		} 
+		}
+	}
+	//TODO consider different mults
+	return count;
+}
+void
+OdinGridAnalyzer::fillGridWithBlockType(std::vector<std::vector<std::pair<char*,int>>>& grid, 
+										t_grid_loc_def* grid_loc_def, 
+										std::vector<t_physical_tile_type> tileTypes,
+										ssize_t indexOfTile,
+										int grid_width,int grid_height)
+	{
+		t_physical_tile_type* type = &(tileTypes[indexOfTile]);
+		/**
+		 * This code is a lightweight version of vpr/src/base/SetupGrid.cpp
+		 */
+		t_formula_data vars;
+		vars.set_var_value("W", grid_width);
+		vars.set_var_value("H", grid_height);
+		vars.set_var_value("w", type->width);
+		vars.set_var_value("h", type->height);
+
+		//Load the x specification
+		auto& xspec = grid_loc_def->x;
+
+		VTR_ASSERT_MSG(!xspec.start_expr.empty(), "x start position must be specified");
```

Comment:
Use the Odin one since the VTR assert as compile levels and might be skipped ( I don't know how they do their assertion levels and I'd rather avoid adding the complexity here unless we drop our own version and change over completely to theirs) keeping both is not something I would do, use the oassert or if guard an error_message